### DISCUSSION TITLE
WPILib Beta 1 is now available

### DISCUSSION THREAD

**Discussion by PeterJohnson** [wpilibsuite/2024Beta#1] [Announcements]
WPILib Beta 1 is now available: https://github.com/wpilibsuite/allwpilib/releases/tag/v2024.1.1-beta-1

Note this requires a 2024 roboRIO image to use on a robot, as well as 2024 beta releases for vendor dependencies.  We expect these to be available soon.

Python is not included in this release, expect a separate announcement soon on that as well.

**Comment by GrahamSH-LLK** [comment]
Will the 2024 roboRIO image be posted here or will it be delivered via an update in FRC Game Tools?

**Reply by ThadHouse** [reply]
It will come through Game Tools, which will be posted here when they're ready.

**Reply by roberttuttle** [reply]
FYI ... our team is running roboRIO 1s (not 2s) and it appears that the Beta 1 game tools only have a v2 image available for flashing.  We were trying to do an end to end test upgrade of our code into 2024 and hit this wall where we could not get our roboRIO 1 updated.

**Reply by ThadHouse** [reply]
Hmm. They should be showing up in the imaging too. 

C:\Program Files (x86)\National Instruments\LabVIEW 2023\project\roboRIO Tool\FRC Images

If you navigate to that folder, what images are there?

**Reply by roberttuttle** [reply]
Here is the error:
<img width="601" alt="image" src="https://github.com/wpilibsuite/2024Beta/assets/512050/9420373d-bad9-4785-8de7-d29313417993">

... and here was the directory looks like (only v2 image in there):
<img width="487" alt="image" src="https://github.com/wpilibsuite/2024Beta/assets/512050/3f307b3b-de4c-4795-b092-d0e6becc6e87">

**Reply by ThadHouse** [reply]
The roborio 1 uses the FRC_roboRIO_2024_v1.1.zip file in the FRC Images folder, not the SD Images folder.

When you go to start the imaging process, what image is selected in the imaging tool?

**Reply by roberttuttle** [reply]
It is "seeing" the roboRIO v1 .zip file in the parent FRC images directory, but still throwing the exception of "cannot open file/folder".
<img width="643" alt="Screenshot 2023-12-08 185251" src="https://github.com/wpilibsuite/2024Beta/assets/512050/ef355818-7dd9-4593-b8f7-ad2d4c60b34a">

**Reply by roberttuttle** [reply]
Note: this is the third computer we have tried with the NI beta 1 install (and several reinstalls of NI game tools on same computer) just to rule out a one-time installer issue.

**Reply by ThadHouse** [reply]
https://docs.wpilib.org/en/stable/docs/software/roborio-info/recovering-a-roborio-using-safe-mode.html

Can you try following the steps here to reimage using safe mode? That error is usually a file system error on the rio side.

**Reply by roberttuttle** [reply]
Sure, we can try again, but we were already in safe mode on the latest attempts.

**Reply by ThadHouse** [reply]
https://knowledge.ni.com/KnowledgeArticleDetails?id=kA03q000000kOHkCAM&l=en-US

If that doesn't work, try following the steps here to fully reformat the roborio. Its kind of difficult to find a flash drive that works, just have to keep trying a bunch. Try as small as you can.

**Reply by roberttuttle** [reply]
Cool, will go down that route next. On a whim, I dug out a "clean" Windows 10 comp and installed NI game tools beta 1 again fresh to rule out Windows 11 as well. Ran as Administrator, roboRIO in safe mode, no firewalls on, etc. and still same error.

**Reply by roberttuttle** [reply]
You were not kidding about the flash drive issue. We are on our 5th different drive and still no go on the recovery. Old school 2GB one we dug out even. Haven't tried the suggestion in the KB article about using a powered USB hub in the mix though.

**Reply by roberttuttle** [reply]
Following up ... just got the Beta 2 NI download and am now able to reformat the roboRIO 1 successfully.
